tests/integration/common/block_utils.rs

```diff
@@ -167,13 +167,23 @@ pub fn os_hints(
     transactions: Vec<InternalTransaction>,
     tx_execution_infos: Vec<TransactionExecutionInfo>,
 ) -> (StarknetOsInput, ExecutionHelperWrapper) {
-    let mut contracts: HashMap<Felt252, ContractState> = blockifier_state
-        .state
-        .address_to_class_hash
-        .keys()
+    let state_diff = blockifier_state.to_state_diff();
+    let deployed_addresses = state_diff.address_to_class_hash.keys().cloned().collect::<HashSet<_>>();
+
+    let initial_addresses = blockifier_state.state.address_to_class_hash.keys().cloned().collect::<HashSet<_>>();
+
+    let addresses = initial_addresses.union(&deployed_addresses);
```

Comment:
The way you compute this is inefficient because you allocate 3 hashsets instead of one. You can do the following instead:

```
let addresses: HashSet<_> = deployed_addresses.keys().zip(initial_addresses.keys()).collect();
```

As you iterate over this again just below, you can even not collect() the iterator at all and just use it directly instead of `addresses.into_iter()`.